src/Maestro/Maestro.MergePolicies/MergePolicy.cs

```diff
@@ -47,6 +49,41 @@ public string Name
 
         public abstract string DisplayName { get; }
 
-        public abstract Task EvaluateAsync(IMergePolicyEvaluationContext context, MergePolicyProperties properties);
+        public abstract Task<MergePolicyEvaluationResult> EvaluateAsync(IPullRequest pr, IRemote darc);
+
+        public Task<MergePolicyEvaluationResult> Pending(string message)
+        {
+            return Task.FromResult(new MergePolicyEvaluationResult(MergePolicyEvaluationStatus.Pending, message, this));
+        }
+
+        public Task<MergePolicyEvaluationResult> Succeed(string message)
+        {
+            return Task.FromResult(new MergePolicyEvaluationResult(MergePolicyEvaluationStatus.Success, message, this));
+        }
+
+        public Task<MergePolicyEvaluationResult> Fail(string message)
+        {
+            return Task.FromResult(new MergePolicyEvaluationResult(MergePolicyEvaluationStatus.Failure, message, this));
+        }
+    }
+
+    public interface IMergePolicy : IMergePolicyInfo
+    {
+        Task<MergePolicyEvaluationResult> EvaluateAsync(IPullRequest pr, IRemote darc);
+    }
+
+    public interface IMergePolicyBuilder
+    {
+        public string Name { get; }
+
+        /// <summary>
+        /// Creates list of instances of concrete merge policies which shall be evaluated 
+        /// for particular merge policy definition
+        /// In most cases it will return array of exactly one merge policy, but in special cases like standard-policy
+        /// it will return multiple pre-configured policies which that policies template consist of
+        /// </summary>
+        /// <param name="properties"></param>
```

Comment:
Remove the param/returns comments. The param comment list isn't complete, and there is nothing useful in the elements anyway.  Bad/empty comments are worse than no comments.